craft_store/publisher/_request.py

```diff
@@ -0,0 +1,41 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Request models for the publisher gateway."""
+
+import re
+from typing import Annotated, TypedDict
```

Comment:
One additional remark about this: before `python3.12`, you should rather use the one from `typing-extensions`:
https://typing-extensions.readthedocs.io/en/latest/index.html#typing_extensions.TypedDict

Reply:
Good to know, thanks!
```suggestion
from typing import Annotated
```